src/Lean/Widget/InteractiveGoal.lean

```diff
@@ -13,15 +13,27 @@ import Lean.Data.Lsp.Extra
 namespace Lean.Widget
 open Server
 
+/-- In the infoview, if multiple hypotheses `h₁`, `h₂` have the same type `α`, they are rendered as `h₁ h₂ : α`.
+We call this a 'hypothesis bundle'. -/
 structure InteractiveHypothesisBundle where
   /-- The user-friendly name for each hypothesis.
   If anonymous then the name is inaccessible and hidden. -/
   names : Array Name
+  /-- The ids for each variable. Should have the same length as `names`. -/
   fvarIds : Array FVarId
   type : CodeWithInfos
+  /-- The value, in the case that the hypothesis is assigned. -/
   val? : Option CodeWithInfos := none
+  /-- The hypothesis is a typeclass instance. -/
   isInstance : Bool
+  /-- The hypothesis is a type. -/
   isType : Bool
+  /-- If true, the hypothesis was not present on the previous tactic state. -/
+  isInserted? : Option Bool := none
```

Comment:
This has three possible values. What is the semantics of `some false`? If the `Option` is just to avoid sending extra data when unnecessary, let's explain in the comment that `some false` should never be used.

Reply:
If I do `isInserted? : Bool`, will the json deriver encode that properly?

Reply:
No, we'd be sending `"isInserted": false` then which takes up space, so the `Option` seems like the right choice; but it seems worth noting that this is the reason for using `Option`.